### ISSUE TITLE
Running this standalone?

### ISSUE THREAD

**Issue by TamaMcGlinn** [georgek/magit-pretty-graph#2]
I'm looking to run this standalone, in order to improve the commandline git log graph; if possible, it would be a good addition [on this SO question](https://stackoverflow.com/questions/1057564/pretty-git-branch-graphs). I want to then [integrate this into a vim git log plugin, vim-flog](https://github.com/rbong/vim-flog/issues/49).

So I tried [these instructions](http://ergoemacs.org/emacs/elisp_running_script_in_batch_mode.html) to run elisp standalone using emacs, but to no avail:

```
magit-pretty-graph$ emacs --script magit-pretty-graph.el 
Loading /etc/emacs/site-start.d/00debian.el (source)...
Loading /etc/emacs/site-start.d/50asymptote.el (source)...
Loading /etc/emacs/site-start.d/50autoconf.el (source)...
Loading /etc/emacs/site-start.d/50cmake-data.el (source)...
Loading /etc/emacs/site-start.d/50devhelp.el (source)...
Loading /etc/emacs/site-start.d/50dictionaries-common.el (source)...
Loading debian-ispell...
Loading /var/cache/dictionaries-common/emacsen-ispell-default.el (source)...
Loading /var/cache/dictionaries-common/emacsen-ispell-dicts.el (source)...
Loading /etc/emacs/site-start.d/50gtk-doc-tools.el (source)...
Loading /etc/emacs/site-start.d/50latex-cjk-common.el (source)...
Loading /usr/share/emacs/site-lisp/latex-cjk-common/cjk-enc.el (source)...
Loading /etc/emacs/site-start.d/50latex-cjk-thai.el (source)...
Loading /etc/emacs/site-start.d/50texlive-lang-english.el (source)...
Cannot open load file: No such file or directory, magit
```

Do you think it is possible to run this script standlone? Otherwise, how would you go about porting the logic; perhaps to common lisp?

**Comment by georgek** [comment]
I think if you wanted to run this standalone it would be better rewritten in something instead of emacs lisp. Emacs has a relatively large startup cost and isn't very suitable for running ad hoc tasks.

Have you looked at [tig](https://jonas.github.io/tig/)? It has a fairly similar output to this.

I could probably rewrite this as a standalone Python script if you think that would be useful.

**Comment by TamaMcGlinn** [comment]
That would be amazing. I am currently using [this modified version of Jan Engelhardt's vim-forest](https://github.com/TamaMcGlinn/dotfiles/blob/master/linuxbash/forest) ([screenshot here](https://github.com/TamaMcGlinn/vim-flogmenu)), but yours looks a lot better ([in the screenshot over here](https://github.com/magit/magit/issues/495)); you've got colours for each branch.

Tig is hard to look at, and needs a lot of configuring. I spent some years trying to configure keybindings so it would be somewhat palatable, but gave up and went back to commandline git eventually. These days, I prefer to stay within vim and leverage a variety of plugins, each managing one part of my git workflow well.

**Comment by georgek** [comment]
I've started to look at this again. Is this still of interest? I had a look and I've seen that you're using git-foresta now. Is there anything you'd want this to do that you can't currently do?

**Comment by TamaMcGlinn** [comment]
Yes. Having one particular commit (defaulting to HEAD) and all its ancestors in colour, while the rest is grey - this helps a lot to find your place in a convoluted mess of merges, and is a feature in GitExtensions.
